### PR TITLE
AudienceRun Bid Adapter: Add Support for User Matching, Referer, …

### PR DESCRIPTION
**Pull request by audiencerun**
## Type of change

- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter   
- [X] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Add Support for User Matching, Referer, Schain, US Privacy, GLVID

## Other information

PR https://github.com/prebid/prebid.github.io/pull/3687

### CODE REVIEW

**Comment by ChrisHuie** [comment]
@audiencerun can you please remove changes to the package-lock file 🙏

**Comment by audiencerun** [comment]
Hi @ChrisHuie, just restored the package-lock.json file. Thanks.

File: modules/audiencerunBidAdapter.js [hunk 1]
```diff
@@ -44,19 +42,53 @@ function getBidFloor(bid) {
...
+    return 0;
   }
 }
 
+/**
+ * Returns the most top page referer.
+ *
+ * @returns {string}
+ */
+function getPageReferer() {
```
Comment (robertrmartinez) [inline modules/audiencerunBidAdapter.js:54]: Is there a reason this is being duplicated?

`bidderRequest.refererInfo.referer` does this logic already, so not really sure this is needed.

https://github.com/prebid/Prebid.js/blob/master/src/refererDetection.js

**Review by robertrmartinez — CHANGES_REQUESTED** [review]

File: modules/audiencerunBidAdapter.js (hunk 1 above)
Comment (audiencerun) [reply modules/audiencerunBidAdapter.js:54]: Hi @robertrmartinez, it is not a duplicate as `bidderRequest.refererInfo.referer` returns only the URL of the page where Prebid.js is called while we need the URL that linked to this page and not the URL of the current page.

Here is a screen that shows it and is available at https://7buoei.csb.app/?pbjs_debug=true

image

Feel free if you need more details.

**Review by robertrmartinez — APPROVED** [review]
We will be making a util function or add this to refererDetection in order to make sure adapters do not duplicate code. Thanks!

**Pull request merged**